src/org/jitsi/impl/neomedia/transform/fec/FlexFec03HeaderReader.java

```diff
@@ -0,0 +1,122 @@
+ * |R|F| P|X|  CC   |M| PT recovery |         length recovery      |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                          TS recovery                          |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |   SSRCCount   |                    reserved                   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                             SSRC_i                            |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |           SN base_i           |k|          Mask [0-14]        |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |k|                   Mask [15-45] (optional)                   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |k|                                                             |
+ * +-+                   Mask [46-108] (optional)                  |
+ * |                                                               |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                     ... next in SSRC_i ...                    |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+public class FlexFec03HeaderReader
+{
+    /**
+     * The minimum size of the FlexFec header, in bytes.
+     * We include here one instance of SSRC_i and SN base_i, as well as
+     * the smallest possible packet mask.
+     */
+    private static int HEADER_MIN_SIZE_BYTES = 20;
+
+    /**
+     * The offset of the mask in the flexfec header, relative to the start
+     * of the flexfec header
+     */
+    private static int MASK_START_OFFSET_BYTES = 18;
+
+    /**
+     * Parse a buffer pointing to FlexFec data.
+     * @param flexFecPacket the (empty) FlexFec03Packet which the parsed
+     * values will be written into
+     * @param buffer buffer which contains the flexfec data
+     * @param flexFecOffset the flexFecOffset in buffer at which the flexfec header starts
+     * @param length length of the buffer
+     * @return true if parsing succeeded, false otherwise
+     */
+    public static boolean readFlexFecHeader(FlexFec03Packet flexFecPacket,
+                                            byte[] buffer, int flexFecOffset,
+                                            int length)
+    {
+        if (length < HEADER_MIN_SIZE_BYTES)
+        {
+            return false;
+        }
+        boolean retransmissionBit = ((buffer[flexFecOffset] & 0x80) >> 7) == 1;
+        if (retransmissionBit)
+        {
+            // We don't support flexfec retransmissions
+            return false;
+        }
+        int maskType = (buffer[flexFecOffset] & 0x40) >> 6;
+        if (maskType != 0)
+        {
+            // We only support flexible (f == 0) mask type
+            return false;
+        }
+
+        int ssrcCount = buffer[flexFecOffset + 8];
```

Comment:
The byte might have a negative value, perhaps add &0xff?

Reply:
nice catch!  fixed and added a unit test to verify